services/movie_session.py

```diff
@@ -0,0 +1,47 @@
+from datetime import datetime
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession
+
+
+def create_movie_session(movie_show_time: str,
+                         movie_id: int,
+                         cinema_hall_id: int) -> None:
+    MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id
+    )
+
+
+def get_movies_sessions(session_date: str = None) -> QuerySet:
+    if session_date:
+        return MovieSession.objects.filter(
+            show_time__date=datetime.strptime(session_date, "%Y-%m-%d").date())
+    return MovieSession.objects.all()
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: str = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None) -> None:
+
+    movie_session = MovieSession.objects.filter(id=session_id)
+
+    if movie_session.exists():
+        if show_time:
+            movie_session.update(show_time=show_time)
+        if movie_id:
+            movie_session.update(movie=movie_id)
+        if cinema_hall_id:
+            movie_session.update(cinema_hall=cinema_hall_id)
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def delete_movie_session_by_id(session_id: int) -> None:
+    MovieSession.objects.filter(id=session_id).delete()

```

Comment:
read the 3 item here
https://github.com/mate-academy/py-movie-moviesession-cinemahall/blob/master/checklist.md#code-efficiency

Reply:
Yes, thanks! I even fixed it, but did not commited! Still need to read the checklist before the review. Guilty!